[ros2] tinyxml2_vendor not in export set

When trying to export targets which depend on pluginlib, CMake complains that `tinyxml2_vendor` is not part of any export set.
I've seen this independent from each other within rosbag2 and image_transport.

```
Starting >>> image_transport
--- stderr: image_transport
CMake Error: install(EXPORT "image_transport" ...) includes target "image_transport" which requires target "tinyxml2_vendor" that is not in any export set.
CMake Generate step failed.  Build files cannot be regenerated correctly.
make: *** [cmake_check_build_system] Error 1
---
Failed   <<< image_transport	[ Exited with code 2 ]
```

Does it make sense to export the vendor package in the first place? I mean, it makes sense for pluginlib to use the vendor package to link correctly to `tinyxml2`, but I am not seeing any reason ad-hoc to also export the vendor package if the `pluginlib` target transitively links correctly against `tinyxml2`.

--- a/pluginlib/CMakeLists.txt
+++ b/pluginlib/CMakeLists.txt
@@ -21,8 +21,9 @@ find_package(TinyXML2 REQUIRED)  # provided by tinyxml2 upstream, or tinyxml2_ve
 add_library(${PROJECT_NAME} INTERFACE)
 target_include_directories(${PROJECT_NAME} INTERFACE
   "$<INSTALL_INTERFACE:include>")
+target_link_libraries(${PROJECT_NAME} INTERFACE tinyxml2::tinyxml2)
 ament_target_dependencies(${PROJECT_NAME} INTERFACE
-  ament_index_cpp class_loader rcutils rcpputils tinyxml2_vendor TinyXML2)
+  ament_index_cpp class_loader rcutils rcpputils)
 ament_export_dependencies(ament_index_cpp class_loader rcutils rcpputils tinyxml2_vendor TinyXML2)
 ament_export_include_directories(include)
 ament_export_targets(${PROJECT_NAME})

--- a/pluginlib/CMakeLists.txt
+++ b/pluginlib/CMakeLists.txt
@@ -21,9 +21,8 @@ find_package(TinyXML2 REQUIRED)  # provided by tinyxml2 upstream, or tinyxml2_ve
 add_library(${PROJECT_NAME} INTERFACE)
 target_include_directories(${PROJECT_NAME} INTERFACE
   "$<INSTALL_INTERFACE:include>")
-target_link_libraries(${PROJECT_NAME} INTERFACE tinyxml2::tinyxml2)
 ament_target_dependencies(${PROJECT_NAME} INTERFACE
-  ament_index_cpp class_loader rcutils rcpputils)
+  ament_index_cpp class_loader rcutils rcpputils TinyXML2)
 ament_export_dependencies(ament_index_cpp class_loader rcutils rcpputils tinyxml2_vendor TinyXML2)
 ament_export_include_directories(include)
 ament_export_targets(${PROJECT_NAME})

--- a/pluginlib/pluginlib-extras.cmake
+++ b/pluginlib/pluginlib-extras.cmake
@@ -40,14 +40,3 @@ if(UNIX AND NOT APPLE)
   # ament_export_libraries() because it is not absolute and cannot be found with find_library
   list(APPEND pluginlib_LIBRARIES ${FILESYSTEM_LIB})
 endif()
-
-find_package(tinyxml2_vendor REQUIRED)
-find_package(TinyXML2 REQUIRED)
-list(APPEND pluginlib_LIBRARIES ${TinyXML2_LIBRARIES})
-
-add_library(tinyxml2_vendor INTERFACE IMPORTED)
-target_include_directories(tinyxml2_vendor INTERFACE
-  ${TinyXML2_INCLUDE_DIRS})
-target_link_libraries(tinyxml2_vendor INTERFACE
-  ${TinyXML2_LIBRARIES})
-#list(APPEND pluginlib_TARGETS tinyxml2_vendor)

--- a/pluginlib/pluginlib-extras.cmake
+++ b/pluginlib/pluginlib-extras.cmake
@@ -50,4 +50,4 @@ target_include_directories(tinyxml2_vendor INTERFACE
   ${TinyXML2_INCLUDE_DIRS})
 target_link_libraries(tinyxml2_vendor INTERFACE
   ${TinyXML2_LIBRARIES})
-list(APPEND pluginlib_TARGETS tinyxml2_vendor)
+#list(APPEND pluginlib_TARGETS tinyxml2_vendor)

